test/testvarid.cpp

```diff
@@ -1472,7 +1479,7 @@ class TestVarID : public TestFixture {
     void varid_cpp_keywords_in_c_code4() { // #12120
         {
             const char code[] = "int false = 0;";
-            ASSERT_THROW_INTERNAL_EQUALS(tokenize(code, dinit(TokenizeOptions, $.cpp = true)), INTERNAL, "Internal error. VarId set for bool literal.");
```

Comment:
I don't understand why we want to allow this. This code looks like invalid C++ code to me. A syntaxError would have been preferable?

Reply:
We don't want to allow it. Currently, we first set the varid and then complain about it. But even setting the varid is wrong in certain situation, see the test case.